src/kem/kyber768/ind_cpa.rs

```diff
@@ -284,28 +284,26 @@ pub(crate) fn decrypt(
     for (i, u_bytes) in
         (0..u_as_ntt.len()).zip(ciphertext.chunks((COEFFICIENTS_IN_RING_ELEMENT * 10) / 8))
     {
-        let u = deserialize_little_endian(10, u_bytes);
+        let u = deserialize_little_endian_10(u_bytes);
         u_as_ntt[i] = ntt_representation(decompress(u, 10));
     }
 
     // v := Decompress_q(Decode_{d_v}(c + d_u·k·n / 8), d_v)
     let v = decompress(
-        deserialize_little_endian(VECTOR_V_COMPRESSION_FACTOR, &ciphertext[VECTOR_U_SIZE..]),
+        deserialize_little_endian_4(&ciphertext[VECTOR_U_SIZE..]),
         VECTOR_V_COMPRESSION_FACTOR,
     );
 
     // sˆ := Decode_12(sk)
-    let mut secret_as_ntt_ring_element_bytes = secret_key.chunks(BITS_PER_RING_ELEMENT / 8);
+    let mut secret_as_ntt_ring_element_bytes = secret_key.chunks(BYTES_PER_RING_ELEMENT);
     for i in 0..secret_as_ntt.len() {
-        secret_as_ntt[i] = deserialize_little_endian(
-            12,
+        secret_as_ntt[i] = deserialize_little_endian_12(
             secret_as_ntt_ring_element_bytes.next().expect("secret_as_ntt_ring_element_bytes should have enough bytes to deserialize to secret_as_ntt"),
```

Comment:
We should be able to use `secret_key.chunks(BYTES_PER_RING_ELEMENT).enumerate()` for the loop now and get rid of the unwrap here.